broker/aws/impl.go

```diff
@@ -126,6 +127,27 @@ func (b *Broker) accountHumanNameFromName(accountName string) (string, error) {
 	return "", errors.New("accountNAme not found")
 }
 
+func (b *Broker) finishUnsealing() error {
+	sessionCredentials, region, err := b.getCredentialsFromProfile(
+		masterAWSProfileName)
+	if err != nil {
+		b.logger.Printf("Unable to get master credentials: %s\n", err)
+		return nil
+	}
+	masterSession, err := session.NewSession(
+		aws.NewConfig().WithCredentials(sessionCredentials).WithRegion(region))
+	if err != nil {
+		return err
+	}
+	if masterSession == nil {
+		return errors.New("masterSession == nil")
+	}
+	b.masterStsClient = sts.New(masterSession)
+	b.masterStsRegion = region
+	b.isUnsealedChannel <- nil
+	return nil
```

Comment:
it was my understanding that even these sessions expire... so thi may stop working after 8 hours... have you tested this in a long running service?

Reply:
I've been running this over the weekend.

Reply:
Instances were up for over 24 hours. I tested both master role (EC2 role) and static credentials master.

Reply:
For EC2 it's actually a credentials provider, which under the covers regenerates them when they are close to expiring.